src/semaphore.c

```diff
@@ -716,6 +771,11 @@ _dispatch_thread_semaphore_create(void)
 	int ret = sem_init(s4, 0, 0);
 	DISPATCH_SEMAPHORE_VERIFY_RET(ret);
 	return (_dispatch_thread_semaphore_t)s4;
+#elif USE_FUTEX_SEM
+	dispatch_futex_t *futex = malloc(sizeof(*futex));
```

Comment:
Hmmm, just got it to run last night with tests passing, though that doesn't mean much.

So three questions to 
a) is the storage for the futex required beyond the life time of a thread?
b) are there multiple instances over the life of the thread (though one at a time)?
c) will any other thread point at it ?
Otherwise, why can't we use thread local?